cumulusci/tasks/release_notes/task.py

```diff
@@ -55,3 +73,127 @@ def _run_task(self):
         self.logger.info("\n" + release_notes)
+
+
+class ParentPullRequestNotes(BaseGithubTask):
+    task_docs = """
+    Aggregate change notes from child pull request(s) to its corresponding
+    parent's pull request.
+    
+    When given the branch_name option, this task will: (1) check if the base branch
+    of the corresponding pull request starts with the feature branch prefix and if so (2) attempt
+    to query for a pull request corresponding to this parent feature branch. (3) if a pull request
+    isn't found, one is created and the BUILD_NOTES_LABEL is applied to it.
+
+    If the BUILD_NOTES_LABEL is present on the pull request, then all notes from the 
+    child pull request are aggregated into the parent pull request. If the BUILD_NOTES_LABEL
+    is not detected on the parent pull request then a link to the child pull request
+    is placed under the "Unaggregated Pull Reqeusts" header.
+
+    When given the parent_branch_name option, this task will query for a corresponding pull request.
+    If a pull request is not found, the task exits. If a pull request is found, then all notes
+    from child pull requests are re-aggregated and the body of the parent is replace entirely.
+    """
+
+    BUILD_NOTES_LABEL = "Build Change Notes"
+
+    task_options = {
+        "branch_name": {"description": "Name of branch with a pull request"},
+        "parent_branch_name": {
+            "description": "name of the parent branch to rebuild change notes for"
+        },
+    }
+
+    def _run_task(self):
+        branch_name = self.options.get("branch_name")
+        parent_branch_name = self.options.get("parent_branch_name")
+
+        if (not branch_name and not parent_branch_name) or (
+            branch_name and parent_branch_name
+        ):
+            raise TaskOptionsError(
+                "You must specify either branch_name or (exclusive) parent_branch_name."
+            )
+
+        self.repo = self.get_repo()
+        generator = ParentPullRequestNotesGenerator(
+            self.github, self.repo, self.project_config
+        )
+
+        if branch_name:
+            self._handle_branch_name_option(generator, branch_name)
+        elif parent_branch_name:
+            self._handle_parent_branch_name_option(generator, parent_branch_name)
+
+    def _handle_branch_name_option(self, generator, branch_name):
+        parent_pull_request = None
+        pull_request = get_pull_request_by_branch_name(self.repo, branch_name)
+
+        if not pull_request:
+            self.logger.info(
+                "Pull request not found for branch: {}.\nExiting...".format(branch_name)
+            )
+            return
+
+        base_branch = pull_request.base.ref
+        feature_prefix = self.project_config.project__git__prefix_feature
+        if not base_branch.startswith(feature_prefix):
+            self.logger.info("Pull request's base is not a feature branch.\nExiting...")
+            return
```

Comment:
We don't need the child pull request in order to find the parent pull request. We can do it from the parent/child branch naming convention:

1. If branch_name doesn't have a double underscore, we can exit early (it's not a child)
2. Split branch_name on double underscore and take the first part as the parent branch name (i.e. starting with feature/parent__child we expect the parent branch name to be feature/parent)
3. Look for a pull request where head = the parent branch name and base is repo.default_branch. This should be the parent pull request.

Reply:
done!